library/Icingadb/Web/Controller.php

```diff
@@ -71,6 +71,14 @@ public function createLimitControl()
         $limitControl = new LimitControl(Url::fromRequest());
         $limitControl->setDefaultLimit($this->getPageSize(null));
 
+        if (Url::fromRequest()->getParam(ViewModeSwitcher::DEFAULT_VIEW_MODE_PARAM) === 'minimal') {
+            $hasLimitParam = Url::fromRequest()->hasParam($limitControl->getLimitParam());
+
+            if ($limitControl->getDefaultLimit() <= LimitControl::DEFAULT_LIMIT && ! $hasLimitParam) {
+                $limitControl->setDefaultLimit($limitControl->getDefaultLimit() * 2);
+            }
+        }
+
```

Comment:
Not sure why you moved this into `createLimitControl`. But in any case, to me this is the incorrect place. It's the view mode switcher why we'd change the default limit. So it's `createViewModeSwitcher`'s duty to do that.

Think of a view that has no view mode switcher but a limit control. Why should the limit be doubled just because there's a `view=minimal` parameter in the url?

Reply:
> Visiting ?view=minimal directly (by not using the switcher) doesn't apply the increased limit

This is why it is necessary here.

Reply:
`createViewModeSwitcher` also runs always. (It doesn't need to be in the success handler.) And has access to the limit control.

Reply:
✅